mita/openstack.py

```diff
@@ -157,47 +157,73 @@ def _get_flavor(self) -> NodeSize:
             logger.error(be)
             raise OpenStackDriverError("Encountered an unknown exception.")
 
-    def _has_free_ip_address(self, network: str) -> bool:
+    def _get_network_by_name(self, name: str) -> OpenStackNetwork:
         """
-        Return True if the given network has an IPv4 Address to lease.
+        Retrieve the OpenStackNetwork instance using the provided name.
+
+        Args:
+            name:   the name of the network.
+
+        Returns:
+            OpenStackNetwork instance
+
+        Raises:
+            ResourceNotFound when no matching resource was found.
+        """
+        url = f"{self.driver_v2._networks_url_prefix}?name={name}"
+        object_ = self.driver_v2.network_connection.request(url).object
+        networks = self.driver_v2._to_networks(object_)
+
+        if not networks:
+            raise ResourceNotFound(f"No network resource with name {name} found.")
+
+        return networks[0]
+
+    def _has_free_ip_address(self, net: OpenStackNetwork) -> bool:
+        """
+        Return True if the given network has an ip address to lease.
 
         Arguments:
-            network: String, The network UUID
+            net:    The reference to OpenStackNetwork
```

Comment:
This is a great change. It makes a lot more sense to me to pass in a full `OpenStackNetwork` object than to pass the UUID. Thanks.

Reply:
Thank you.